applications/FluidDynamicsApplication/python_scripts/navier_stokes_embedded_solver.py

```diff
@@ -278,13 +344,24 @@ def Initialize(self):
                                                                                                number_of_avg_nodes)
 
         # If required, intialize the FM-ALE utility
-        if (self.settings["fm_ale_settings"]["fm_ale_step_frequency"].GetInt() != 0):
+        if (self.settings["fm_ale_settings"]["fm_ale_step_frequency"].GetInt() > 0):
             self.fm_ale_step = 1
             # Fill the virtual model part geometry. Note that the mesh moving util is created in this first call
-            self._get_mesh_moving_util().FillVirtualModelPart(self.main_model_part)
+            self._get_mesh_moving_util().Initialize(self.main_model_part)
 
         KratosMultiphysics.Logger.PrintInfo("NavierStokesEmbeddedMonolithicSolver", "Solver initialization finished.")
 
+    def AdvanceInTime(self, current_time):
+        # Call base solver AdvanceInTime to clone the time step and get the new time
+        new_time = super(NavierStokesEmbeddedMonolithicSolver, self).AdvanceInTime(current_time)
+
+        # Save the current step and time in the virtual model part process info
+        if self.settings["fm_ale_settings"]["fm_ale_step_frequency"].GetInt() > 0:
+            self._get_fm_ale_virtual_model_part().ProcessInfo[KratosMultiphysics.STEP] += 1
+            self._get_fm_ale_virtual_model_part().ProcessInfo[KratosMultiphysics.TIME] = new_time
```

Comment:
if you could use the same processinfo for the virtualModelPart then this would not be needed I think

Reply:
True. This is how it used to be. However, I prefer to have different ProcessInfo containers for each one of the model parts.